src/styles/calendar-block.scss

```diff
@@ -0,0 +1,39 @@
+.calendar {
+  width: 728px;
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: flex-start;
+  gap: $gap;
+  padding: 10px;
+  margin: 20px auto;
+
+  &__day {
+    @extend %font-style;
+    @extend %square-style;
+    cursor: pointer;
+    transition-duration: 0.5s;
+  }
+
+  @for $index from 1 through 31 {
+    &__day--#{$index}::before {
```

Comment:
Why do you need modifiers `calendar__day--#{$index}`. Try to rewrite it with `&:nth-child`.

This way you'll avoid redundant modifiers.